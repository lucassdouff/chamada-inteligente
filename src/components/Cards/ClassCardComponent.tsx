import { useNavigation } from "@react-navigation/native";
import { StackNavigationProp } from "@react-navigation/stack";
import { TouchableWithoutFeedback } from "react-native";
import { View, Text } from "react-native";
import { UserClassesDTO } from "../../core/dtos/UserClassesDTO";

interface ClassCardComponentProps {
    userClass: UserClassesDTO;
    staticMode?: boolean;
    schedule?: Date;
}

export default function ClassCardComponent({userClass, staticMode, schedule}: ClassCardComponentProps) {

    const navigation = useNavigation<StackNavigationProp<any>>();
    
    return (
        <TouchableWithoutFeedback onPress={staticMode ? undefined : () => navigation.navigate('Turma', {userClass: userClass})} >
            <View className="h-24 flex flex-col px-3 py-2 border border-gray-400 rounded mb-3 bg-blue-600">
                <View className="flex-row">
                    <Text className="text-white text-lg">{userClass.code + " - "}</Text>
                    <Text className="text-white text-lg">{userClass.name}</Text>
                </View>
                <Text className="text-white mb-2">{userClass.semester}</Text>
                {staticMode ? 
                <Text className="text-white">{schedule?.toString()}</Text>
                : null}
            </View>
        </TouchableWithoutFeedback>
        
    )
}