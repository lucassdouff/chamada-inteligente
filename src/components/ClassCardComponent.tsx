import { useNavigation } from "@react-navigation/native";
import { StackNavigationProp } from "@react-navigation/stack";
import { Button } from "react-native";
import { View, Text } from "react-native";

interface ClassBoxComponentProps {
    codigoTurma: string;
    nomeTurma: string;
    semestre: string;
}

export default function ClassBoxComponent({codigoTurma, nomeTurma, semestre}: ClassBoxComponentProps) {

    const navigation = useNavigation<StackNavigationProp<any>>();
    
    return (
        <View className="flex flex-col px-3 py-5 border border-gray-400 rounded mb-3">
            <View className="flex-row mb-2">
                <Text>{codigoTurma + " - "}</Text>
                <Text>{nomeTurma}</Text>
            </View>
                <Text>{semestre}</Text>
            <Button title="ACESSAR TURMA" onPress={() => navigation.navigate('Turma')} />
        </View>
    )
}