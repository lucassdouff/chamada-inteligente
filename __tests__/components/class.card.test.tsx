import React from 'react';
import ClassCardComponent from '../../src/components/Cards/ClassCardComponent';
import { fireEvent, render, screen, waitFor } from '@testing-library/react-native';
import '@testing-library/jest-dom';

const mockNavigate = jest.fn();

jest.mock("@react-navigation/native", () => {
    const actualNav = jest.requireActual("@react-navigation/native");
    return {
      ...actualNav,
      useNavigation: () => ({
        navigate: mockNavigate,
        dispatch: jest.fn(),
      }),
    };
});

jest.mock('../../src/core/controllers/NavigationController', ()=>{
  const actualNav = jest.requireActual('../../src/core/controllers/NavigationController');
  return {
      ...actualNav,
      navigationController: () => ({
          setSession: jest.fn(),
      }),
  };
});

describe('ClassCard component should render correctly', () => {
    
    it('should render correctly', () => {

        render(<ClassCardComponent userClass={{
          id_class: 1,
          name: 'Class 1',
          code: '123456',
          semester: '2021-1',
          id_teacher: 1,
          id_course: 1,
          class_weekdays: [{
              weekday: 1,
              start_hour: '08:00:00',
              end_hour: '10:00:00'
          }]
        }} />);
        
        expect(screen.getByText("123456 - Class 1")).toBeTruthy();
        expect(screen.getByText("2021-1")).toBeTruthy();
        
    });

    it('should render correctly with schedule', () => {
            
            render(<ClassCardComponent staticMode userClass={{
            id_class: 1,
            name: 'Class 1',
            code: '123456',
            semester: '2021-1',
            id_teacher: 1,
            id_course: 1,
            class_weekdays: [{
                weekday: 1,
                start_hour: '08:00:00',
                end_hour: '10:00:00'
            }]
            }} schedule={[{
            weekday: 1,
            start_hour: '08:00:00',
            end_hour: '10:00:00'
            }]}/>);
            
            expect(screen.getByText("123456 - Class 1")).toBeTruthy();
            expect(screen.getByText("2021-1")).toBeTruthy();
            expect(screen.getByText("Segunda - 08:00 Ã s 10:00")).toBeTruthy();
    });

    it('should redirect to class screen when clicking on card', async () => {
        
        render(<ClassCardComponent userClass={{
          id_class: 1,
          name: 'Class 1',
          code: '123456',
          semester: '2021-1',
          id_teacher: 1,
          id_course: 1,
          class_weekdays: [{
              weekday: 1,
              start_hour: '08:00:00',
              end_hour: '10:00:00'
          }]
        }} />);
        
        fireEvent.press(screen.getByText(/123456 - Class 1/i));
        
        await waitFor(()=>{
            expect(mockNavigate).toHaveBeenNthCalledWith(1  , "Turma", {
                userClass: {
                    class_weekdays: [
                        {
                            end_hour: "10:00:00",
                            start_hour: "08:00:00",
                            weekday: 1
                        }
                    ],
                    code: "123456",
                    id_class: 1,
                    id_course: 1,
                    id_teacher: 1,
                    name: "Class 1",
                    semester: "2021-1"
                }
            });
        })

    });
});
