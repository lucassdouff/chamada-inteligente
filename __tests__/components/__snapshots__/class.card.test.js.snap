// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClassCard component should render correctly renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "height": 96,
      },
      {
        "display": "flex",
      },
      {
        "flexDirection": "column",
      },
      {
        "paddingLeft": 12,
        "paddingRight": 12,
      },
      {
        "paddingBottom": 8,
        "paddingTop": 8,
      },
      {
        "borderBottomWidth": 1,
        "borderLeftWidth": 1,
        "borderRightWidth": 1,
        "borderTopWidth": 1,
      },
      {
        "borderBottomColor": "#9ca3af",
        "borderLeftColor": "#9ca3af",
        "borderRightColor": "#9ca3af",
        "borderTopColor": "#9ca3af",
      },
      {
        "borderBottomLeftRadius": 4,
        "borderBottomRightRadius": 4,
        "borderTopLeftRadius": 4,
        "borderTopRightRadius": 4,
      },
      {
        "marginBottom": 12,
      },
      {
        "backgroundColor": "#2563eb",
      },
    ]
  }
  testID="class-card"
>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "width": "80%",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#fff",
            },
            {
              "fontSize": 18,
              "lineHeight": 28,
            },
          ]
        }
      >
        123456 - Class 1
      </Text>
    </View>
  </View>
  <Text
    style={
      [
        {
          "color": "#fff",
        },
        {
          "marginBottom": 8,
        },
      ]
    }
  >
    2021-1
  </Text>
</View>
`;
